patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -3664,15 +3664,17 @@
  			if (!Lighting.NotRetro) {
  				oldDrawWater(bg, Style, Alpha);
  				return;
-@@ -44801,7 +_,7 @@
+@@ -44801,8 +_,8 @@
  						float num5 = rand.Next((int)num3, (int)num4);
  						num5 *= (1f - maxRaining + 1f) / 2f;
  						num5 *= (1f - windSpeedTarget + 1f) / 2f;
 -						num5 *= (float)dayRate;
+-						if (rand.Next((int)num5) == 0)
 +						num5 *= (float)desiredWorldEventsUpdateRate;
- 						if (rand.Next((int)num5) == 0)
++						if (rand.NextFloat() <= 1f / num5)
```

Comment:
This code is weird. It looks like, in vanilla, the faster the day is going, the rarer lightning gets?
Unfortunately that means if someone sets `desiredWorldEventsUpdateRate` small enough here, then you'll get lightning every frame...